app/bundles/EmailBundle/Swiftmailer/Transport/AmazonApiTransport.php

```diff
@@ -0,0 +1,842 @@
+                    if ($evt) {
+                        $evt->setResult(\Swift_Events_SendEvent::RESULT_SUCCESS);
+                        $evt->setFailedRecipients($failedRecipients);
+                        $this->getDispatcher()->dispatchEvent($evt, 'sendPerformed');
+                    }
+                },
+                'rejected' => function (AwsException $reason, $iteratorId) use ($commands, $evt) {
+                    $this->triggerSendError($evt, []);
+                },
+            ]);
+            $promise = $pool->promise();
+            $promise->wait();
+
+            return count($commands);
+        } catch (\Exception $e) {
+            $this->triggerSendError($evt, $failedRecipients);
+            $message->generateId();
+            $this->throwException($e->getMessage());
+        }
+
+        return 1;
+    }
+
+    /**
+     * @param \Swift_Mime_Message $message
+     *
+     * @return array
+     */
+    public function getAmazonMessage(\Swift_Mime_Message $message)
+    {
+        $this->message = $message;
+        $metadata      = $this->getMetadata();
+        $emailBody     = $this->message->getBody();
+
+        if (!empty($metadata)) {
+            $metadataSet  = reset($metadata);
+            $tokens       = (!empty($metadataSet['tokens'])) ? $metadataSet['tokens'] : [];
+            $mauticTokens = array_keys($tokens);
+        }
+
+        foreach ($metadata as $recipient => $mailData) {
+            $this->message->setBody($emailBody);
+            $msg        = $this->messageToArray($mauticTokens, $mailData['tokens'], true);
+            $rawMessage = $this->buildRawMessage($msg, $recipient);
+            $payload    = [
+                'Source'       => $msg['from']['email'],
+                'Destinations' => [$recipient],
+                'RawMessage'   => [
+                    'Data' => $rawMessage,
+                    ],
+            ];
+
+            yield $payload;
+        }
+    }
+
+    /**
+     * @param type $msg
+     * @param type $recipient
+     *
+     * @return string
+     */
+    public function buildRawMessage($msg, $recipient)
+    {
+        $separator           = md5(time());
+        $separator_multipart = md5($msg['subject'].time());
+        $message             = "MIME-Version: 1.0\n";
+        $message .= 'Subject: '.$msg['subject']."\n";
+        $message .= 'From: '.$msg['from']['name'].' <'.$msg['from']['email'].">\n";
+        $message .= "To: $recipient\n";
+        if (count($msg['recipients']['cc']) > 0) {
+            $message .= 'Cc: '.implode(',', array_keys($msg['recipients']['cc']))."\n";
+        }
+        if (count($msg['recipients']['bcc']) > 0) {
+            $message .= 'Bcc: '.implode(',', array_keys($msg['recipients']['bcc']))."\n";
+        }
+        $message .= "Content-Type: multipart/mixed; boundary=\"$separator_multipart\"\n";
```

Comment:
if (isset($msg['headers'])) {
          foreach ($msg['headers'] as $key => $value) {
            $message .= "$key: " . $value . "\n";
          }
        }

Reply:
This snippet allow this function to send Mautic custom headers and the unsubscribe url

Reply:
Hope that somone can find a solusion to add the headers to the sendBulkTemplatedEmail method